opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.hpp

```diff
@@ -114,13 +281,42 @@ namespace Opm {
 
     */
 
+    //namespace {
+
+    class RestartSchedule {
+        /*
+           The content of this struct is logically divided in two; either the
+           restart behaviour is governed by { timestep , basic , frequency }, or
+           alternatively by { rptshec_restart_set , rptsched_restart }.
+
+           The former triplet is mainly governed by the RPTRST keyword and the
+           latter pair by the RPTSCHED keyword.
+           */
+        public:
 
+            RestartSchedule() = default;
+            RestartSchedule( size_t sched_restart);
+            RestartSchedule( size_t step, size_t b, size_t freq);
+            bool writeRestartFile( size_t timestep , const TimeMap& timemap) const;
+            bool operator!=(const RestartSchedule& rhs) const;
+            bool operator==( const RestartSchedule& rhs ) const;
+
+
+
+        //private:
```

Comment:
Yes, eventually. The experimental implementation was easier this way, and the full interface needs some work, which will be done once @joakim-hove returns.

In short, `RestartConfig` should be "overridable" by applications, i.e. for simulations the preferred way of overriding deck specified configurations should be to provide your own `RestartConfig` instance, rather than the ugly `override*` calls we use now. Since that will break this interface anyway, re-privatising these vars can be done in the same sweep.